Sources/ComposableArchitecture/Scheduling/TestScheduler.swift

```diff
@@ -34,9 +34,9 @@ where SchedulerTimeType: Strideable, SchedulerTimeType.Stride: SchedulerTimeInte
     let delta = self.now.distance(to: nextDate)
     self.now = nextDate
 
-    while let (_, date, action) = self.scheduled.first, date == nextDate {
+    while let (id, _, date, action) = self.scheduled.first, date == nextDate {
       action()
-      self.scheduled.removeFirst()
+      self.scheduled.removeAll(where: { $0.id == id })
```

Comment:
`removeFirst` was not correct here because in the process of executing `action()` we may trigger the work's cancellable (as is the case with `debounce`), which will remove the work from the `scheduled` queue, which will then cause us to accidentally remove the wrong work from the queue.